sql/information_schema/information_schema.go

```diff
@@ -1863,6 +1913,8 @@ func (t *informationSchemaTable) Schema() Schema {
 }
 
 func (t *informationSchemaTable) AssignCatalog(cat Catalog) Table {
```

Comment:
These changes are kind of nonsensical

It doesn't matter if you prevent a data race in assigning the catalog to these table objects if there is one global copy of each that gets used across all sessions / queries. Sure you are preventing literal races, but it's still the case that e.g. the same table object in use by one session about to start returning rows can have its catalog object re-assigned by another session. The mutex needs to guard the unit of work, which is the transaction. It's just not a workable solution in this context.

What you need to do return a new copy of each table every time when asked, rather than keeping a single global instance of each.

You can avoid doing this (I think) if you just back out the parallel changes for now.